src/PassPlanFilePlugin.groovy

```diff
@@ -0,0 +1,71 @@
+import static TerraformEnvironmentStage.PLAN
+import static TerraformEnvironmentStage.APPLY
+
+class PassPlanFilePlugin implements TerraformPlanCommandPlugin, TerraformApplyCommandPlugin, TerraformEnvironmentStagePlugin {
+
+    public static void init() {
+        PassPlanFilePlugin plugin = new PassPlanFilePlugin()
+
+        TerraformEnvironmentStage.addPlugin(plugin)
+        TerraformPlanCommand.addPlugin(plugin)
+        TerraformApplyCommand.addPlugin(plugin)
+    }
+
+    @Override
+    public void apply(TerraformEnvironmentStage stage) {
+        stage.decorate(PLAN,  archivePlanFile(stage.getEnvironment()))
+        stage.decorate(APPLY, downloadArchive(stage.getEnvironment()))
+    }
+
+    @Override
+    public void apply(TerraformPlanCommand command) {
+        String env = command.getEnvironment()
+        command.withArgument("-out=tfplan-" + env)
+    }
+
+    @Override
+    public void apply(TerraformApplyCommand command) {
+        String env = command.getEnvironment()
+        command.withDirectory("tfplan-" + env)
+    }
+
+    public Closure archivePlanFile(String env) {
+        return { closure ->
+            closure()
+            echo "Archiving tfplan-${env} file"
+            archiveArtifacts artifacts: "tfplan-" + env
```

Comment:
Is there a reason to use `archiveArtifacts` vs `stash`?

Reply:
Nope, I changed it to stash